### ISSUE TITLE
Allow "documents" to show in Block Editor link suggestions

### ISSUE THREAD

**Issue by benklocek** [wp-document-revisions/wp-document-revisions#258]
As the title states, I was hoping to be able to see "document" CPTs in the link suggestions in the Block Editor, but they were not showing.

I was hoping for a simple solution and found it in adjusting the post type args. I believe this should be default on, but in case others want it, here's what I used to allow it.

Example: https://www.loom.com/share/77120bf51b09450eaaf18b575f231ed6

```
/** Adjust settings on Document CPT for access in link suggestions */
add_filter( 'register_post_type_args', 'xxx_change_document_post_type_args', 999, 2 );
function xxx_change_document_post_type_args( $args, $post_name ) {
    if ( $post_name != 'document' ) return $args;
   
    $args['show_in_rest'] = true; //show in Block Editor link suggestions

    return $args;
}
```

**Comment by NeilWJames** [comment]
@benklocek,
Thanks for this. in fact there is a specific plugin filter that deals with this allowing simpler coding:
```
<?php
/** Adjust settings on Document CPT for access in link suggestions */
add_filter( 'document_revisions_cpt', 'xxx_change_document_post_type_args' );
function xxx_change_document_post_type_args( $args ) {
    $args['show_in_rest'] = true; //show in Block Editor link suggestions

    return $args;
}
```

Of course, the primary purpose of the parameter is to open the rest interface to documents - and you can't get one withour the other.

There is an amount of code that hides the actual file name of the document on the server. With document being defined in the interface, it becomes easier to do this decode process (without requiring authorisation).

Having come to help support this plugin some years after its creation, I don't know whether this hiding process is a current requirement. I would ask Ben Balter to say whether this functionality should be implemented.

However I should point out that the delivered WP rest functionality will provide this information, although it is more tedious to do the mapping.

It may be possible to amend the output of the rest interface to maintain this difficulty if the requirement is still current. The ability to do this needs to be investigated.

@benbalter,
Thoughts please on whether this should be implemented and whether to also try to modify the rest interface for documents. 

Regards,
Neil James

**Comment by benbalter** [comment]
> Thoughts please on whether this should be implemented and whether to also try to modify the rest interface for documents.

If I'm understanding correctly, without modification, turning on the REST interface would allow a user with access to a document to see the true document path, not the permalink? If so, I believe that would violate user expectations, as (1) the document can then be shared without authentication, and (2) subsequent access changes (e.g., making it private) would not actually change access to the document if someone had the upload URL.

I'm supportive of API support, but wouldn't want to add it at the expense of our authorization model, at least not without making sure we're not giving the admin a hidden foot gun if they activate the feature.

**Comment by benklocek** [comment]
I think I get what you are describing, and don't have an opinion about the access though the REST interface as our use case is simply using it to access through the link suggestions, which does insert the permalink, not the true document path.

I just hope we don't need to lose access to that if you need to tighten security.

**Comment by NeilWJames** [comment]
@benklocek,
To answer the question of the last paragraph, the post itself contains a link to the document - and contains just the post title.

The API access route is via the post id. Both of these routes will do a security validation before serving the post (including if the post id used is that of the attachment). So the post is secure.

[The only complaint that I personally would have is that the error message is similar to "Not Authorised" rather than "Not Found". So it tells you that there is a post with the id.]

Hope this helps,
Neil

**Comment by NeilWJames** [comment]
@benklocek,
I have created a Pull request which when released will provide a REST interface for documents.

However you will need to enable it by entering the code
`add_filter( 'document_show_in_rest', '__return_true' );`
in your code.

Neil James

**Comment by NeilWJames** [comment]
Addressed by #265
